hubblestack_nova/modules/cve_scan_v2.py

```diff
@@ -0,0 +1,283 @@
+'''
+
+:maintainer: HubbleStack
+:maturity: 20160214
+:platform: Linux
+:requires: SaltStack
+
+Using the vulners.com api, this cve scan module checks your machine
+for packages with known and reported vulnerabilities.
+
+Future addition:
+    Using salt:// url as the source of the vulnerabilites
+    Using a more efficient caching method so we just add on new
+        vulnerabilities and not re-query an entire year back
+    Restructure loop to go through local pkgs and check if in known
+        vulnerabilties, not vice versa like it is now
+
+
+Sample YAML data
+
+cve_scan_v2:
+    ttl: 86400
+    url: http://vulners.com/api/v3/search/lucene/
+'''
+from __future__ import absolute_import
+import logging
+
+import salt
+import salt.utils
+import requests
+import copy
+import re
+
+import json
+import os
+
+from distutils.version import LooseVersion
+from datetime import datetime
+from time import time as current_time
+
+log = logging.Logger(__name__)
+
+def __virtual__():
+    return True
+
+def audit(data_list, tags, verbose=False):
+    global cache_path
+    os_name = __grains__['os'].lower() 
+    cache_path = '/var/cache/salt/minion/files/base/cve/%s.json' % (os_name)
+    cache = {}
+    #Make cache directory and all parent directories
+    # if it doesn't exist.
+    if not os.path.exists(os.path.dirname(cache_path)):
+        os.makedirs(os.path.dirname(cache_path))
+  
+    # Go through yaml to check for cve_scan_v2,
+    #    if its present, check for a cached version 
+    #    of the scan.
+    for data in data_list:
+
+        if 'cve_scan_v2' in data:
+
+            ttl = data['cve_scan_v2']['ttl']
+            url = data['cve_scan_v2']['url']
+            # Requests only handles http:// requests   
+            if url.startswith('https'):
+                url.replace('https', 'http', 1)
```

Comment:
OK, can you please open a ticket (here on Github) to look into salt.utils.http so we remember to get this working over https?